### DISCUSSION TITLE
what are some good techniques for using HTMX with a maps api (google maps, leaflet, etc.)?

### DISCUSSION THREAD

**Discussion by trawn3333** [bigskysoftware/htmx#2495] [Q&A]
Are there any suggested ways of interacting with a maps API instance with HTMX? Or are SPA-style tools generally more the way to go? Maybe a mix of the two?

**Comment by andrewgryan** [comment]
> Are there any suggested ways of interacting with a maps API instance with HTMX? Or are SPA-style tools generally more the way to go? Maybe a mix of the two?

There was a great project called [Hyperleaflet](https://github.com/cemrehancavdar/hyperleaflet) that went a long way to implementing this in a HTMX compatible way. The original author archived the repo recently, but I have a [fork](https://github.com/andrewgryan/hyperleaflet) where I've been implementing new features in a **develop** branch. I've released it on npm under `@andrewgryan/hyperleaflet` because I feel this is too good an idea to let it vanish.

I'm hoping to reduce the amount of wrapping code needed to patch missing features by making a one to one mapping between leaflet constructor options and data attributes. The tricky bit will be the change handlers needed to update layers on the map.
